### PR TITLE
Add default Anthropic autocomplete model

### PR DESCRIPTION
**Pull request by justinmilner1**
Fixes https://github.com/sourcegraph/cody/issues/4639

There are two 'model' parameters - one was being set to a default when undefined, the other was not. 
This PR adds a default for that undefined parameter.

## Test plan
- Replicated the error (use anthropic setting: "cody.autocomplete.advanced.provider": "anthropic")
- Observe that autocomplete works on Anthropic setting with this addition
- Error described in above bug is not present

### CODE REVIEW

**Comment by justinmilner1** [comment]
@valerybugakov Hey Valery - this is my first PR to Cody, looking for a review - thanks! (tagging you because you worked on this section most recently)

File: vscode/src/completions/providers/anthropic.ts [hunk 1]
```diff
@@ -296,7 +296,7 @@ export function createProviderConfig({
                     ...providerOptions,
                     id: PROVIDER_IDENTIFIER,
                 },
-                { maxContextTokens, model, ...otherOptions }
+                { maxContextTokens, model: model ?? 'anthropic/claude-instant-1.2', ...otherOptions }
```
Comment (valerybugakov) [inline vscode/src/completions/providers/anthropic.ts:299]: Let's update the model value for the `providerConfig` on line 304. 

```ts
const clientModel = model ?? 'anthropic/claude-instant-1.2'

// Use this variable in both places
```

Then, we will need to update tests with the new value: `vscode/src/completions/providers/create-provider.test.ts`

**Review by valerybugakov — COMMENTED** [review]
Thank you, @justinmilner1! 

I verified that the fix works locally. Could you follow up on the inline comment I left? Then we can merge this.

**Comment by justinmilner1** [comment]
@valerybugakov Made the review suggestion and updated tests. 

Unit tests passed - except for unrelated enterprise level tests.

**Comment by valerybugakov** [comment]
@justinmilner1, thanks, I missed the notification!

**Review by valerybugakov — APPROVED** [review]
Thank you! 🙌

**Pull request merged**